server/core/uno.py

```
@@ -0,0 +1,215 @@
+        top_card = random.choice(self.remaining_cards)
+        while top_card.is_special():
+            top_card = random.choice(self.remaining_cards)
+
+        self.game_stack: List[Card] = [top_card]
+
+    def remove_player(self, player) -> None:
+        self.players.remove(player)
+
+    def validate_players(self) -> None:
+        if len(self.players) < self.MIN_PLAYERS_ALLOWED:
+            raise Exception(f"need at least {self.MIN_PLAYERS_ALLOWED} players to start the game")
```

Ruff (0.13.2)

Create your own exception

(TRY002)